src/backend/media-route-handlers.ts

```diff
@@ -16,3 +29,73 @@ export const mediaProxy = createProxyMiddleware({
   pathFilter: (path) => path.startsWith('/media'),
   pathRewrite: { '^/media': '' },
 })
+
+const s3Client = new S3Client({
+  region,
+  credentials: {
+    // fallback to '' for now so it does not fail in CI
+    accessKeyId: process.env.BUCKET_ACCESS_KEY_ID ?? '',
+    secretAccessKey: process.env.BUCKET_SECRET_ACCESS_KEY ?? '',
+  },
+  endpoint,
+  forcePathStyle: true, // test, maybe only set on dev
+})
+
+const mimeTypeDecoder = t.union([
+  t.literal('image/gif'),
+  t.literal('image/jpeg'),
+  t.literal('image/png'),
+  t.literal('image/svg+xml'),
+  t.literal('image/webp'),
+])
+
+export async function mediaPresignedUrl(req: Request, res: Response) {
+  const mimeType = decodeURIComponent(String(req.query.mimeType))
+  if (!mimeTypeDecoder.is(mimeType)) {
+    res.status(400).send('Missing or invalid mimeType')
+    return
+  }
+
+  const editorVariant = decodeURIComponent(String(req.query.editorVariant))
+  if (!t.string.is(req.query.editorVariant)) {
+    // if we have this code in the same monorepo as the editor we can check known editorVariant values
+    res.status(400).send('Missing or invalid editorVariant')
+    return
+  }
+
+  const editorHost = req.headers.host
+  if (!t.string.is(editorHost) || !editorHost.length) {
+    res.status(400).send('Missing header: host')
+  }
+
+  const fileHash = createId() // cuid since they are shorter and look less frightening
+
+  const variantFolder = editorVariant === 'unknown' ? 'all' : editorVariant
+
+  const [mediaType, mediaSubtype] = mimeType.split('/')
+  const fileExtension = mimeType === 'image/svg+xml' ? 'svg' : mediaSubtype
+
+  // Keys with slashes are expected in S3 (rendered as folders in bucket view)
+  const fileName = `${variantFolder}/${fileHash}/${mediaType}.${fileExtension}`
```

Comment:
Yes it is. From the Miro:
> **Type**: we use the first part of the mimeType (e.g. image, audio, video) as the filename so when users download the file they get `image.png` instead of `{weirdHash}.png`. we could also use i18n versions or alternatives if we want at a later point.